src/optimizer/index_util.cpp

```diff
@@ -48,9 +48,15 @@ bool IndexUtil::SatisfiesSortWithIndex(
     if (tv_col_oid == mapped_cols[idx_ind]) {
       // Column is present in both sort and index so increment both
       sort_ind++, idx_ind++;
-    } else if (bounds != nullptr && bounds->find(lookup[mapped_cols[idx_ind]]) != bounds->end()) {
-      // If column is found in bounds but not sort, continue
-      idx_ind++;
+    } else if (bounds != nullptr) {
+      auto bound_column = bounds->find(lookup[mapped_cols[idx_ind]]);
+      if (bound_column != bounds->end() && bound_column->second[0] == bound_column->second[1]) {
```

Comment:
From my understanding (let me know if this is incorrect) if the first N continuous columns of the index are not in the sort, then they don't have to be bound exactly. It's just the inner columns in the index but not in the sort that need to be bound exactly. I'm not sure how important it is to capture this scenario though.

Reply:
I don't think that's always true:
c1 | c2 | c3
 1  |  3   | 4
 1  |  3   | 5
 2  |  3   | 4
 
 If you want the elements sorted by c2, c3 then since the first column isn't bound, the last tuple wouldn't be in the correct sort order (the inner loop runs faster than the outer loop is my way of thinking about it).

Reply:
Ah now I get it, thanks for the explanation